### PR TITLE
Minor Zephyr port features

### PR DESCRIPTION
**Pull request by jadonk**
* modzsensor: include a more complete list of sensor type constants
* modusocket: add `recvfrom` function useful for broadcast UDP messages
* ~mdns_service: add ability to broadcast mdns advertisements~

### CODE REVIEW

File: ports/zephyr/modusocket.c [hunk 1]
```diff
@@ -293,6 +293,35 @@ STATIC mp_obj_t socket_recv(mp_obj_t self_in, mp_obj_t len_in) {
...
+    socklen_t slen = sizeof(saddr);
+
+    ssize_t recv_len = zsock_recvfrom(socket->ctx, vstr.buf, max_len, 0, (struct sockaddr *)&saddr, &slen);
+    if (recv_len == -1) {
+        vstr_clear(&vstr);
+        mp_raise_OSError(errno);
+    }
+
+    if (recv_len == 0) {
+        return mp_const_empty_bytes;
```
Comment (dpgeorge) [inline ports/zephyr/modusocket.c:315]: I think in this case it should still return a tuple, like `(b"", <addr>)`.

**Comment by dpgeorge** [comment]
Thanks for the contribution!  Apart from the above comment, these changes look good.

**Review by MaureenHelm — APPROVED** [review]

**Comment by projectgus** [comment]
This is an automated heads-up that we've just merged a Pull Request
that removes the STATIC macro from MicroPython's C API.

See https://github.com/micropython/micropython/pull/13763

A search suggests this PR might apply the STATIC macro to some C code. If it
does, then next time you rebase the PR (or merge from master) then you should
please replace all the `STATIC` keywords with `static`.

Although this is an automated message, feel free to @-reply to me directly if
you have any questions about this.

File: ports/zephyr/modusocket.c (hunk 1 above)
Comment (dpgeorge) [reply ports/zephyr/modusocket.c]: I made this changed during merge.

**Pull request merged**